# **Risk Factors of Sleep Disorders: Analyzing the Impact of Gender, Occupation, and Stress Levels on Sleep Duration and Sleep Disorders**

**Overview**

Using a real-world data-set, this project will look into how different factors, including a person's gender, occupation, and stress levels may correlate with sleep duration and sleep quality. Specifically, it aims to uncover:



*   Differences in sleep duration across a variety of professions and stress levels
*  Correlations between stress and the likelihood of developing a sleep disorder
*  Gender-based trends in sleep duration, stress levels, and sleep disorders

**Why Does This Matter?**

Sleep health is a significant predictor of both mental and physical well-being. By understanding the lifestyle and demographic factors associated with decreased sleep duration and sleep disorders, we can effectively tailor sleep improvement strategies and provide preventative care in high-risk groups.




In [ ]:
import pandas as pd

from google.colab import files

uploaded = files.upload()

In [ ]:
import pandas as pd

df = pd.read_csv("Sleep_health_and_lifestyle_dataset.csv")
df.head()

Person ID Gender  Age            Occupation  Sleep Duration  \
0          1   Male   27     Software Engineer             6.1   
1          2   Male   28                Doctor             6.2   
2          3   Male   28                Doctor             6.2   
3          4   Male   28  Sales Representative             5.9   
4          5   Male   28  Sales Representative             5.9   

   Quality of Sleep  Physical Activity Level  Stress Level BMI Category  \
0                 6                       42             6   Overweight   
1                 6                       60             8       Normal   
2                 6                       60             8       Normal   
3                 4                       30             8        Obese   
4                 4                       30             8        Obese   

  Blood Pressure  Heart Rate  Daily Steps Sleep Disorder  
0         126/83          77         4200            NaN  
1         125/80          75        10000            NaN  
2         125/80          75        10000            NaN  
3         140/90          85         3000    Sleep Apnea  
4         140/90          85         3000    Sleep Apnea

In [ ]:
df["Sleep Disorder"].value_counts()

Sleep Disorder
Sleep Apnea    78
Insomnia       77
Name: count, dtype: int64

In [ ]:
df_clean = df.dropna(subset=["Sleep Disorder"])

In [ ]:
df.isnull().sum()

Person ID                    0
Gender                       0
Age                          0
Occupation                   0
Sleep Duration               0
Quality of Sleep             0
Physical Activity Level      0
Stress Level                 0
BMI Category                 0
Blood Pressure               0
Heart Rate                   0
Daily Steps                  0
Sleep Disorder             219
dtype: int64

# Stress Level and Sleep Duration

In [ ]:
df.groupby("Stress Level")["Sleep Duration"].mean()

Stress Level
3    8.226761
4    7.030000
5    7.483582
6    7.454348
7    6.468000
8    6.050000
Name: Sleep Duration, dtype: float64

There is a negative correlation between stress levels and sleep duration.

# Stress Level and Occupation

In [ ]:
df.groupby("Occupation")["Stress Level"].mean()

Occupation
Accountant              4.594595
Doctor                  6.732394
Engineer                3.888889
Lawyer                  5.063830
Manager                 5.000000
Nurse                   5.547945
Sales Representative    8.000000
Salesperson             7.000000
Scientist               7.000000
Software Engineer       6.000000
Teacher                 4.525000
Name: Stress Level, dtype: float64

Sales Representatives, Salespersons, and Scientists have the highest average stress levels (7-8 hours, respectively), while Engineers, Teachers, and Accountants report the lowest stress levels (about 3.89-4.59 hours)

# Stress Level and Gender

In [ ]:
df.groupby("Gender")["Stress Level"].mean()

Gender
Female    4.675676
Male      6.079365
Name: Stress Level, dtype: float64

On average, females (4.68) report lower stress levels than men (6.08).

# Stress Level and Sleep Disorder

In [ ]:
df.groupby("Sleep Disorder")["Stress Level"].mean()

Sleep Disorder
Insomnia       5.870130
Sleep Apnea    5.666667
Name: Stress Level, dtype: float64

# Occupation and Sleep Duration

In [ ]:
df.groupby("Occupation")["Sleep Duration"].mean().sort_values()

Occupation
Sales Representative    5.900000
Scientist               6.000000
Salesperson             6.403125
Teacher                 6.690000
Software Engineer       6.750000
Manager                 6.900000
Doctor                  6.970423
Nurse                   7.063014
Accountant              7.113514
Lawyer                  7.410638
Engineer                7.987302
Name: Sleep Duration, dtype: float64

On average, Sales Representatives, Scientists, and Salespersons sleep the least (about 5.90 - 6.40 hours), while Accountants, Lawyers, and Engineers sleep the most (7.11 - 7.99 hours).

# Occupation and Gender

In [ ]:
pd.crosstab(df["Gender"], df["Occupation"])

Occupation  Accountant  Doctor  Engineer  Lawyer  Manager  Nurse  \
Gender                                                             
Female              36       2        32       2        1     73   
Male                 1      69        31      45        0      0   

Occupation  Sales Representative  Salesperson  Scientist  Software Engineer  \
Gender                                                                        
Female                         0            0          4                  0   
Male                           2           32          0                  4   

Occupation  Teacher  
Gender               
Female           35  
Male              5

Sales and legal roles were mostly occupied by male participants, while nursing, accounting, and teaching roles had more female representation. Additionally, doctor roles were commonly held by males in this dataset.

# Occupation and Sleep Disorder

In [ ]:
pd.crosstab(df["Occupation"], df["Sleep Disorder"])

Sleep Disorder        Insomnia  Sleep Apnea
Occupation                                 
Accountant                   7            0
Doctor                       3            4
Engineer                     5            1
Lawyer                       2            3
Nurse                        3           61
Sales Representative         0            2
Salesperson                 29            1
Scientist                    0            2
Software Engineer            1            0
Teacher                     27            4

Nurses have a very high number of sleep apnea cases (61), while Salespersons and Teachers have the highest numbers of insomnia cases (29 and 27).

# Gender and Sleep Duration

In [ ]:
df.groupby("Gender")["Sleep Duration"].mean()

Gender
Female    7.229730
Male      7.036508
Name: Sleep Duration, dtype: float64

# Gender and Sleep Disorder

In [ ]:
df.groupby("Gender")["Sleep Disorder"].value_counts()

Gender  Sleep Disorder
Female  Sleep Apnea       67
        Insomnia          36
Male    Insomnia          41
        Sleep Apnea       11
Name: count, dtype: int64

Females represent a higher numbers of sleep disorder cases (103) than males (52).

# Key Findings

**Stress Level and Sleep Duration:**
On average, participants with higher stress levels reported shorter sleep hours, revealing an expected negative correlation between stress and sleep quality.

**Occupation & Sleep Duration:** Occupations such as Sales Representative and Scientist had the lowest average sleep durations (5.9 and 6.0 hrs, respectively), while Engineers and Lawyers had the highest (around 8 hrs). These results suggest how job structure and overall workload may significantly influence sleep.

**Occupation & Stress Level:**
Stress varied by profession, with Sales Representatives (8/10) and Scientists (7/10) reporting the highest average stress, while Engineers (3.9/10) and Teachers (4.5/10) having the lowest. This pattern, when paired with the sleep duration data, supports a strong link between occupational stress and sleep reduction.

**Occupation & Gender:** Male participants were mostly represented in legal, sales, and medical professions (e.g., lawyers, salespeople, and doctors), which were also associated with higher stress levels. Female participants were more commonly found in nursing, teaching, and accounting, which revealed moderate-to-lower stress levels.

**Occupation & Sleep Disorders:** Nurses had an exceptionally high count of sleep apnea cases (61), possibly due to irregular shifts and physically demanding schedules. Teachers and Salespeople reported higher cases of insomnia (27 and 29 respectively), highlighting how emotional or cognitive workload may affect sleep quality.

**Gender & Stress Level:** Men (6.1/10) reported slightly higher average stress levels than women (4.7) in this dataset; this aligns with their occupational representation, as men were, on average, in professions with higher reported stress levels.

**Gender & Sleep Disorders:** Despite lower stress, women experienced more sleep disorders (insomnia or apnea) than men, indicating that factors beyond perceived stress, such as biological, hormonal, or psychological difference,—may influence sleep health. This could also be partially explained by occupational placement: a large number of women were represented in the nursing profession, which had the highest number of sleep apnea cases overall.






# Conclusion

This project analyzed a real-world dataset from Kaggle titled "Sleep Health and Lifestyle Dataset," exploring how factors like stress, occupation, and gender relate to sleep duration and sleep disorders. While higher stress was generally linked to shorter sleep, the presence of sleep disorders did not consistently align with stress levels alone. For example, teachers and nurses, both female-dominated roles, reported relatively low average stress but had some of the highest rates of insomnia and sleep apnea. In other words, women experienced more sleep disorders than men despite reporting lower average stress, suggesting how factors like work schedules, job demands, and biological differences may play a major role in sleep health rather than gender or stress level alone.

Overall, these results highlight gender and occupational patterns in sleep disorders, supporting the development of tailored interventions based on job-related risk. A key limitation, however, is that the number of participants across occupations was uneven, which may affect how representative some of the findings are. Future research with a more balanced sample could  provide stronger generalizability, offering more robust insights into the relationships between gender, occupation, stress, and sleep.

